### ISSUE TITLE
Remove the  "powershell tools for visual studio are available" 

### ISSUE THREAD

**Issue by SimonCropp** [microsoft/poshtools#585]
I have other tools for debugging PS and dont want to constantly have the notification there if i am never going to install that extension.

Also it is showing nepotism if no other VS extension can register to get a similar notification for any other file extension.

**Comment by bmoore-msft** [comment]
@SimonCropp - this notification was a "point in time" feature to bootstrap availability of the tools in the pre-release version of VS2015 and CTP5.  We'll be removing the VSIX notification for RTM as the tools are integrated into VS Setup.

**Comment by SimonCropp** [comment]
this is not satisfactory.  it still shows up in VS2013 and i have chosen not to install that feature of vs2015 so still get the annoyance. 

please remove the adware.

**Comment by adamdriscoll** [comment]
@SimonCropp I realize it might be annoying but if you are editing PowerShell scripts enough in VS that you find it annoying, why not install the PoshTools to help you edit those PowerShell scripts?

**Comment by SimonCropp** [comment]
i like to run a lean ide. this includes only adding things that add compelling value. 

i actually tried the PS tools in VS2015. and i didnt like the flicker of a command prompt that showed every time i opened VS

but my usage aside. forcing people to install a tool that is optional via the use of an annoying notice is not a good experience. you even let people close that notice but re-show it every time a ps1 is opened

**Comment by SimonCropp** [comment]
lets look at this from a different perspective. I want to write an optional vs extension for .md files. please tell me how i can get MS to ship a "please install my addin since you opened a .md file" warning inside the next updates for VS 2013 and 2015.

**Comment by adamdriscoll** [comment]
I understand your point but I was just curious as to why you were encountering this notification so frequently, which indicates to me that you are editing PowerShell files in VS, yet you don't want to install PoshTools. I realize you want to run lean and are annoyed by both the pop up and the cmd prompt flicker. 

I'm just trying to understand what we can do to make the PoshTools experience better so that it wouldn't be a big deal if it was just installed. A rich, integrated language experience in VS is what makes it useful so I'm trying to figure out if it's a shortcoming in the implementation. 

Re: MD file extension - This just happens to have been adopted by Microsoft as their solution to the PowerShell language story much like Python Tools is the extension for Python development in VS.

**Comment by SimonCropp** [comment]
I regularly view ps1 files in VS. But when I edit them j use different tooling.

**Comment by adamdriscoll** [comment]
Syntax highlighting isn't helpful at all when viewing? Or is it not worth the extension to have that functionality?

**Comment by SimonCropp** [comment]
@adamdriscoll not really. Basically a feature/tool like this should sell itself. Tell people how awesome it is and let them decide to opt in. Dont annoy them with an annoying prompt and then justify the annoyance with "but we are awesome"

**Comment by SimonCropp** [comment]
@bmoore-msft @adamdriscoll  either way this should be re-opened

**Comment by SimonCropp** [comment]
bump. again please stop trying to install software people dont want

**Comment by BobbyCannon** [comment]
I agree that this notice should be removed. Still getting this note in VS17 (15.3.1)

**Comment by BobbyCannon** [comment]
@bmoore-msft You said "We'll be removing the VSIX notification for RTM as the tools are integrated into VS Setup."

I assume it was never removed?

**Comment by SimonCropp** [comment]
Given this is not fixed can you please re-open it

**Comment by bmoore-msft** [comment]
This is on the backlog...

**Comment by BobbyCannon** [comment]
It's very annoying. I have no desire to install another extension and I've closed that bar like 100 times this week... getting old. @bmoore-msft, is this something that will be add to a sprint soonish? /crossesfingers

**Comment by vbfox** [comment]
I think I yielded on most of my VS installs and just installed the ext but it's an unnecessary burden just to remove a yellow bar.

**Comment by NickCraver** [comment]
Please fix this, I'm dismissing it constantly and it's beyond annoying. There simply can't be a sane argument for having **both** a prompt to install and "you should just install it". It's either in box or it isn't, a constant, repetitive prompt makes *no* sense at all.

If I want it, I'll install it. If I dismiss it, I don't want it. Stop asking me. It's infuriating that this isn't remembered. This is labelled as a question, but it's not. The fact that this preference is not remembered in a bug and should be treated as such.

My father taught me early on: don't ask the question if you don't care about the answer, it's rude. That's exactly what this prompt does every time I open a `.ps1`.

**Comment by SimonCropp** [comment]
2.5 year later. can this please be fixed. just search for the warning message and delete the code that pops it up.

**Comment by adamdriscoll** [comment]
I found it. I will let the team know where to look. 

![image](https://user-images.githubusercontent.com/1256531/30324605-40f0ba48-9787-11e7-8955-b9335fe2877f.png)

**Comment by NickCraver** [comment]
Since comments are better written here than only Twitter conversations, some observations that this is a symptom of a larger problem:

The current state of this PowerShell tools is perplexing to me, it's in some sort of half-adopted state by Microsoft and that's leaking into the user experience. 
1. The tool is in the Microsoft organization here on GitHub, but on the marketplace is ([under a person](https://marketplace.visualstudio.com/items?itemName=AdamRDriscoll.PowerShellToolsforVisualStudio2017-18561), not [the Microsoft organization](https://marketplace.visualstudio.com/search?term=publisher%3A%22Microsoft%22&target=VS&category=All%20categories&vsVersion=&sortBy=Relevance) or any team).
2. The prompt is in Visual Studio unlike *any* other extension I'm aware of (so very much a special case), yet it's not included in the installer.
3. When the user is prompted to install (the notification we're complaining about here), it takes them to the marketplace, not the Visual Studio installer.

This is very confusing on many angles. I propose Microsoft either:
1. Actually adopt this extension and make it part of the installer.
or
2. Leave it *completely* separate the same way every other extension is: remove the prompt.

Bundling it in the installer makes more sense to me. If Microsoft is pushing PowerShell on other platforms and is trying to improve the installer for their main use cases, then having it an option I can script to install like all others seems like the logical option. If that's not the intent, then place remove the prompt completely. It's annoying, weird, confusing, and a one-off all at the same time.

**Comment by adamdriscoll** [comment]
Until Microsoft handles this, I think I found a workaround. 

It requires installing an extension that I just created. I tested on a Win10 VS2017 15.3 box and the gold bar it not shown. It will be available on the Marketplace as soon as that syncs. It's only 5 KB and does nothing more than override one of the exported types in the MSFT Azure PowerShell VSIX to return null rather than the implementation that shows the gold bar. 

https://github.com/adamdriscoll/poshtools.goldbarremover/releases/tag/v1.1
